shared/desktop/webpack.config.babel.js

```diff
@@ -0,0 +1,320 @@
+  isVisDiff: getenv.boolish('VISDIFF', false),
+}
+console.log('Flags: ', flags)
+
+// The common config all other derive from
+const makeCommonConfig = () => {
+  const makeRules = () => {
+    const fileLoaderRule = {
+      loader: 'file-loader',
+      options: {
+        name: '[name].[ext]',
+      },
+    }
+
+    const babelRule = {
+      loader: 'babel-loader',
+      options: {
+        // Have to do this or it'll inherit babelrcs from the root and pull in things we don't want
+        babelrc: false,
+        cacheDirectory: true,
+        plugins: [
+          ['babel-plugin-transform-builtin-extend', {globals: ['Error']}],
+          'transform-flow-strip-types',
+          'transform-object-rest-spread', // not supported by electron yet
+          'babel-plugin-transform-class-properties', // not supported by electron yet
+          'transform-es2015-destructuring', // due to a bug: https://github.com/babel/babel/pull/5469
+        ],
+        presets: [
+          [
+            'env',
+            {
+              debug: false,
+              exclude: ['transform-regenerator'],
+              modules: false,
+              targets: {
+                electron: '1.6.10',
+              },
+              useBuiltIns: false,
+            },
+          ],
+          'babel-preset-react',
+        ],
+      },
+    }
+
+    return [
+      {
+        // Don't include large mock images in a prod build
+        include: path.resolve(__dirname, '../images/mock'),
+        test: /\.jpg$/,
+        use: [flags.isDev ? fileLoaderRule : 'null-loader'],
+      },
+      {
+        include: path.resolve(__dirname, '../images/icons'),
+        test: /\.(flow|native\.js|gif|png|jpg)$/,
+        use: ['null-loader'],
+      },
+      {
+        exclude: /((node_modules\/(?!universalify|fs-extra))|\/dist\/)/,
+        test: /\.jsx?$/,
+        use: [babelRule],
+      },
+      {
+        test: [/emoji-datasource.*\.(gif|png)$/, /\.ttf$/],
+        use: [fileLoaderRule],
+      },
+      {
+        test: /\.css$/,
+        use: ['style-loader', 'css-loader'],
+      },
+    ]
+  }
+
+  const makeCommonPlugins = () => {
+    const defines = {
+      __DEV__: flags.isDev,
+      __HOT__: JSON.stringify(flags.isHot),
+      __SCREENSHOT__: flags.isVisDiff,
+      __VERSION__: flags.isDev ? JSON.stringify('Development') : JSON.stringify(process.env.APP_VERSION),
+      'process.env.NODE_ENV': flags.isDev ? JSON.stringify('development') : JSON.stringify('production'),
+    }
+    console.warn('Injecting defines: ', defines)
+    const definePlugin = [new webpack.DefinePlugin(defines)]
+
+    const uglifyPlugin = flags.isDev
+      ? []
+      : [
+          new UglifyJSPlugin({
+            comments: false,
+            sourceMaps: true,
+          }),
+        ]
+
+    return [...definePlugin, ...uglifyPlugin].filter(Boolean)
+  }
+
+  // If we use the hot server it pulls in this config
+  const devServer = {
+    compress: true,
```

Comment:
doesn't seem to make a noticable difference either way.  i can turn it off. its just gzip compression in express so it shouldn't really be slow